aspnetcore/blazor/blazor-server-ef-core.md

```diff
@@ -0,0 +1,126 @@
+
+## Use of DbContext in Blazor Server Apps
+
+The following recommendations are designed to provide a consistent approach to using EF Core in Blazor Server apps. 
+
+* By default, consider using one context per operation. The context is designed for fast, low overhead instantiation:
+
+    ```csharp
+    var using context = new MyContext();
+    return await context.MyEntities.ToListAsync();
+    ```
+* Use a flag to prevent multiple concurrent operations:
+
+    ```csharp
+    if (Loading)
+    {
+        return;
+    }
+    try 
+    {
+        Loading = true;
+        // operations go here
+    }
+    finally 
+    {
+        Loading = false;
+    }
+    ```
+* For longer-lived operations that take advantage of EF Core's [change tracking](https://docs.microsoft.com/ef/core/querying/tracking) and/or [concurrency control](https://docs.microsoft.com/ef/core/saving/concurrency), scope the context to the lifetime of the component.
+
+### Create new DbContext instances
+
+The fastest way to create a new `DbContext` instance is by using `new` to create a new instance. There are several scenarios that may require additional dependencies to be resolved. For example, you may wish to use [DbContextOptions](https://docs.microsoft.com/ef/core/miscellaneous/configuring-dbcontext#configuring-dbcontextoptions) to configure the context. One approach is to register the options in DI and inject them into the component, like this:
+
+```razor
+@inject DbContextOptions<MyContext> Options
+
+@code {
+    public Task LoadAsync(int id)
+    {
+        using var context = new MyContext(Options);
+        return context.MyEntities.SingleOrDefaultAsync(e => e.Id == id);
+    }
+}
+```
+
+Another solution is to use a factory. EF Core 5.0 (preview) provides a built-in factory for creating new contexts. Because the sample app is based on the latest stable versions and not the preview, it implements its own factory in `Data/DbContextFactory.cs`. 
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Data/DbContextFactory.cs)]
+
+The factory is injected into components and used to create new instances. For example, in `Pages/Index.razor`:
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/Index.razor?range=199-212)]
+
+### Scope to the component lifetime
+
+There are two approaches to scope a context to the component lifetime. If the context is registered with DI, you can use the special `OwningComponentBase` base class to create a control scope. Learn more about the class [here](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.components.owningcomponentbase-1).
+
+The second approach is to use the factory to create a context and track it for the lifetime of the component. First, implement `IDisposable` and inject the factory as shown in `Pages/EditContact.razor`.
```

Comment:
How do we want users to choose between these two approaches? Do we recommend one over the other? If yes, then should we just cover the approach that we recommend? I'm just trying to avoid confusing the user with choices that they might not know how to make. I'm not sure myself which pattern is preferred.